tests/topotests/bgp_evpn_gr/test_bgp_evpn_gr.py

```diff
@@ -0,0 +1,752 @@
+
+# pylint: disable=C0413
+from lib import topotest
+from lib.topogen import Topogen, TopoRouter, get_topogen
+from lib.topolog import logger
+from lib.common_config import (
+    check_router_status,
+    kill_router_daemons,
+    start_router_daemons,
+)
+
+def build_topo(tgen):
+    # Create PE and host routers
+    for name in ["PE1", "PE2", "host1", "host2"]:
+        tgen.add_router(name)
+    # Links: PE1-PE2, PE1-host1, PE2-host2
+    tgen.add_link(tgen.gears["PE1"], tgen.gears["PE2"], "PE1-eth0", "PE2-eth0")
+    tgen.add_link(tgen.gears["PE1"], tgen.gears["host1"], "PE1-eth1", "host1-eth0")
+    tgen.add_link(tgen.gears["PE2"], tgen.gears["host2"], "PE2-eth1", "host2-eth0")
+
+def setup_module(mod):
+    tgen = Topogen(build_topo, mod.__name__)
+    tgen.start_topology()
+
+    # Configure host MACs
+    host_macs = {"host1": "1a:2b:3c:4d:5e:61", "host2": "1a:2b:3c:4d:5e:62"}
+    for name, mac in host_macs.items():
+        host = tgen.net[name]
+        host.cmd_raises(f"ip link set dev {name}-eth0 down")
+        host.cmd_raises(f"ip link set dev {name}-eth0 address {mac}")
+        host.cmd_raises(f"ip link set dev {name}-eth0 up")
+
+    # Configure PE devices: vrf-blue, vxlan100/1000, bridges and sysctls
+    pe_suffix = {"PE1": "1", "PE2": "2"}
+    for name, suf in pe_suffix.items():
+        pe = tgen.net[name]
+        vtep_ip = f"10.100.0.{suf}"
+        bridge_ip = f"50.0.1.{suf}/24"
+        bridge_ipv6 = f"50:0:1::{suf}/48"
+        pe.cmd_raises("ip link add vrf-blue type vrf table 10")
+        pe.cmd_raises("ip link set dev vrf-blue up")
+        pe.cmd_raises(f"ip link add vxlan100 type vxlan id 100 dstport 4789 local {vtep_ip}")
+        pe.cmd_raises("ip link add name br100 type bridge stp_state 0")
+        pe.cmd_raises("ip link set dev vxlan100 master br100")
+        pe.cmd_raises(f"ip link set dev {name}-eth1 master br100")
+        pe.cmd_raises(f"ip addr add {bridge_ip} dev br100")
+        pe.cmd_raises("ip link set up dev br100")
+        pe.cmd_raises("ip link set up dev vxlan100")
+        pe.cmd_raises(f"ip link set up dev {name}-eth1")
+        pe.cmd_raises("ip link set dev br100 master vrf-blue")
+        pe.cmd_raises(f"ip -6 addr add {bridge_ipv6} dev br100")
+        pe.cmd_raises(f"ip link add vxlan1000 type vxlan id 1000 dstport 4789 local {vtep_ip}")
+        pe.cmd_raises("ip link add name br1000 type bridge stp_state 0")
+        pe.cmd_raises("ip link set dev vxlan1000 master br100")
+        pe.cmd_raises("ip link set up dev br1000")
+        pe.cmd_raises("ip link set up dev vxlan1000")
+        pe.cmd_raises("ip link set dev br1000 master vrf-blue")
+        pe.cmd_raises("sysctl -w net.ipv4.ip_forward=1")
+        pe.cmd_raises("sysctl -w net.ipv6.conf.all.forwarding=1")
+
+    # Load FRR configuration and start routers
+    for name, router in tgen.routers().items():
+        router.load_config(TopoRouter.RD_ZEBRA, os.path.join(CWD, f"{name}/zebra.conf"))
```

Comment:
Please use frr.conf (instead of per daemon configs).

Reply:
Done

Reply:
Not done?